M2/Macaulay2/m2/lists.m2

```diff
@@ -28,6 +28,7 @@ List + List  := List => (v,w) -> apply(v,w,plus)
      - List  := List => v -> apply(v,minus)
 List - List  := List => (v,w) -> apply(v,w,difference)
 Thing * List := List => (a,v) -> apply(v,x->a * x)
+List ** List := List => (X,Y) -> flatten for x in X list apply(Y, y -> {x, y})
```

Comment:
If this is implementing a Cartesian product of two lists somehow, then should the result be a list of pairs (of class Sequence), rather than a list of lists of length 2?  What is the intended application?

Reply:
This is used a few places. I'm ambivalent about returning lists or sequences, but it's current applications (for instance in PrimaryDecomposition package) might rely on returning a list of lists. @moorewf what do you prefer?

Reply:
It should be compatible with the way it works for sets:
```m2

i3 : set {a,b,c} ** set {1,2,3}

o3 = set {(a, 1), (a, 2), (a, 3), (b, 1), (b, 2), (b, 3), (c, 1), (c, 2), (c, 3)}

o3 : Set
```

Reply:
Done.